implement [DA1] More of that Strange Oil

tiera3 opened the pull request.

xenohedron commented on Mage.Sets/src/mage/sets/UnknownEvent.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    private UnknownEvent() {
+        super("Unknown Event", "DA1", ExpansionSet.buildDate(2023, 2, 15), SetType.PROMOTIONAL);
```
I believe the set type needs to be JOKE_SET to not be legal in eternal formats. PROMOTIONAL is for fancy reprints. @theelk801 can you confirm?

Grath replied:
Yes, this should be a JOKE_SET like MysteryBoosterPlaytest and the Un-sets.

xenohedron commented on Mage.Sets/src/mage/cards/m/MoreOfThatStrangeOil.java:
```diff
@@ -0,0 +1,59 @@
+        // Choose one --
+        // • Proliferate. Draw a card.
+        this.getSpellAbility().addEffect(new ProliferateEffect());
+        this.getSpellAbility().addEffect(new DrawCardSourceControllerEffect(1).concatBy(". "));
```
no concatBy here. text gen handles that automatically

xenohedron commented on Mage.Sets/src/mage/cards/m/MoreOfThatStrangeOil.java:
```diff
@@ -0,0 +1,59 @@
+public final class MoreOfThatStrangeOil extends CardImpl {
+
+    private static final FilterSpell filter = new FilterSpell("creature, artifact, or planeswalker spell");
+    private static final Predicate<MageObject> predicate = Predicates.or(
```
no point in declaring this predicate as a field, just add it directly where it's used in the static block below

xenohedron commented on Mage.Sets/src/mage/cards/m/MoreOfThatStrangeOil.java:
```diff
@@ -0,0 +1,59 @@
+        this.getSpellAbility().addEffect(new ProliferateEffect());
+        this.getSpellAbility().addEffect(new DrawCardSourceControllerEffect(1).concatBy(". "));
+
+        // * Counter target creature or enchantment spell.
```
fix text comment to match the card

xenohedron commented on Mage.Sets/src/mage/cards/m/MoreOfThatStrangeOil.java:
```diff
@@ -0,0 +1,59 @@
+
+        // Choose one --
+        // • Proliferate. Draw a card.
+        this.getSpellAbility().addEffect(new ProliferateEffect());
```
boolean param false to hide reminder text

xenohedron commented on Mage.Sets/src/mage/cards/m/MoreOfThatStrangeOil.java:
```diff
@@ -0,0 +1,59 @@
+        // * Counter target creature or enchantment spell.
+        Mode mode = new Mode(new CounterTargetEffect());
+        mode.addTarget(new TargetSpell(filter));
+        this.getSpellAbility().addMode(mode);
```
need to add `withFlavorWord()` to the mode

and above, `this.getSpellAbility().withFirstModeFlavorWord()` for the first mode

xenohedron commented on Mage.Sets/src/mage/cards/MoreOfThatStrangeOil.java:
```diff
@@ -0,0 +1,60 @@
+package mage.cards.m;
```
file in wrong location - not sure how you keep doing this

tiera3 commented:
Comments attached to the commit (that went to the wrong place)

removed concatBy
removed predicate declaration
corrected comment (also noticed that I had omitted the Scry 1 at the end)
added param false to ProliferateEffect
added withFlavorWord and withFirstModeFlavorWord
added in the ScryEffect

Does DrawCardSourceControllerEffect and CounterTargetEffect also need a false parameter?
